### ISSUE TITLE
Algorithms are registered on the universe but never deregistered

### ISSUE THREAD

**Issue by bruttenberg** [p2t2/figaro#658]
When a ProbQueryAlgorithm is created, it is registered on the universe, but when a kill is issued, it is never deregistered. This is a problem when you have many nested inference algorithms. This really should be done at the algorithm level automatically for every algorithm created.

**Comment by wkretschmer** [comment]
Is this a duplicate of #596?

**Comment by bruttenberg** [comment]
Indeed, this is a repeat. So this was never fixed in Algorithm, correct?

+  +  +  +  +  +  +  +  +  +  +  +  +  +
Brian Ruttenberg, PhD
Senior Scientist
Decision Management Systems Division
Government Services
Charles River Analytics Inc.
617.491.3474 x730
www.cra.com<http://www.cra.com/>

**Comment by wkretschmer** [comment]
No, we haven't fixed this yet.
